### PR TITLE
Add points change

### PR DESCRIPTION
**Pull request by numel007**
### Checklist
- [x] No merge conflicts (pull from `dev` if needed)
- [ ] Thoroughly tested code
- [x] Trello card requirements are met

### What's new?
* Calculate market_status based off index's market and current time
* Rectified incorrect point/percent change calculations
* Now returns index's market under key `market`

### Known Issues
*  Needs to be tested while market is open and switching from open to close.

### CODE REVIEW

File: src/robinhood.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def check_login_time(cls):
 
         time_delta = abs(datetime.strptime(cls.login_time, format) - datetime.strptime(current_time, format))
 
-        # Relogin after 22 hours
```
Comment (pondorasti) [inline src/robinhood.py]: We should probably undo this change, the term `log` is usually used in the context of saving information for crashes or status reports.

File: src/robinhood.py [hunk 2]
```diff
@@ -71,25 +69,43 @@ def get_ticker(cls, ticker):
...
         ticker_data = {}
+        ticker_data["market"] = instrument_data[0]["market"].strip("https://api.robinhood.com/markets/")
         ticker_data["current"] = round(float(r.stocks.get_latest_price(ticker)[0]),2)
         ticker_data["volume"] = round(float(fundamentals[0]["volume"]),2)
         ticker_data["avg_volume"] = round(float(fundamentals[0]["average_volume"]), 2)
         ticker_data["market_cap"] = round(float(fundamentals[0]["market_cap"]), 2)
-        ticker_data["market_status"] = 0
-        ticker_data["name"] = r.stocks.find_instrument_data(ticker)[0]['simple_name']
-        ticker_data['points_change'] = 0
+        ticker_data["name"] = r.stocks.find_instrument_data(ticker)[0]["simple_name"]
```
Comment (pondorasti) [inline src/robinhood.py:85]: If we are not going to use this variable on the front-end, there's no need to add it to the JSON payload. You can just save it to a variable.

File: src/robinhood.py [hunk 3]
```diff
@@ -71,25 +69,43 @@ def get_ticker(cls, ticker):
...
+        price_change["percent"] = round(float(price_change["points"] / ticker_data["range"]["open"] * 100),2)
+        ticker_data["points_change"] = price_change
+
+        # Market status calculation
+        market_hours = r.get_market_hours(ticker_data["market"], datetime.utcnow().strftime("%Y-%m-%d"))
+        open_time = (lambda y: datetime.strptime(y, "%Y-%m-%d %H:%M:%S") )((lambda x: x.strip("Z").replace("T", " ") )(market_hours["opens_at"]))
+        close_time = (lambda y: datetime.strptime(y, "%Y-%m-%d %H:%M:%S") )((lambda x: x.strip("Z").replace("T", " ") )(market_hours["closes_at"]))
+        now = datetime.utcnow()
+
+        if now >= open_time and now <= close_time:
```
Comment (pondorasti) [inline src/robinhood.py]: We also need to check if the market is open or not. If `is_open=False` then the `opens_at` and `closes_at` values will be null. Is the lambda function going to behave correctly?

File: src/robinhood.py [hunk 4]
```diff
@@ -71,25 +69,43 @@ def get_ticker(cls, ticker):
...
             "date": fundamentals[0]["market_date"]
         }
-
         ticker_data["symbol"] = fundamentals[0]["symbol"]
         ticker_data["timestamp"] = (datetime.utcnow()).strftime("%H:%M:%S")
 
+        # Calculate points and percent change
+        price_change = {}
+        price_change["points"] = round(float(ticker_data["current"] - ticker_data["range"]["open"]),2)
+        price_change["percent"] = round(float(price_change["points"] / ticker_data["range"]["open"] * 100),2)
```
Comment (pondorasti) [inline src/robinhood.py]: Apparently the way we calculate price change for points and percent is wrong.

According to this [article](https://pocketsense.com/calculate-anexpected-growth-rate-using-constant-growth-3970.html):

> Subtract the initial share price of the stock from the ending share price of the stock to find the change in price. For example, if the stock started the day at $59.90 and ended the day at $59.60, subtract $59.90 from $59.60 to get a change of -30 cents, meaning you had a loss of 30 cents.
>
> Divide the change in the stock price by the initial price per share. In this example, divide the loss of 30 cents by the initial stock price of $59.90 to get about 0.005.
>
> Move the decimal place in 0.005 two places to the right to find the stock lost 0.5 percent of its value for the day.

**Review by pondorasti — CHANGES_REQUESTED** [review]

File: src/robinhood.py (hunk 2 above)
Comment (numel007) [reply src/robinhood.py:85]: Isn't the name used on the frontend to display the corp name below the stock index?

Comment (pondorasti) [reply src/robinhood.py:85]: I accidentally highlighted the wrong line, I am referring to the `market` key.

File: src/robinhood.py (hunk 3 above)
Comment (numel007) [reply src/robinhood.py]: Fixed in next commit

File: src/robinhood.py (hunk 2 above)
Comment (numel007) [reply src/robinhood.py:85]: Removed from json in next commit

**Review by pondorasti — APPROVED** [review]
Works like a charm!
